### PR TITLE
Add Pangea4 host configs

### LINKED ISSUE #3172: [EPIC] Update TTE builds
## Description
Update builds on TTE HPC with new machines and modules

## Work Scope
Need to update/clean host-configs and CI Job

## Issues in this EPIC
- [x] #3081 + [TPL-262](https://github.com/GEOS-DEV/thirdPartyLibs/pull/262) Create Pangea4 host-configs
- [x] [TPL-295](https://github.com/GEOS-DEV/thirdPartyLibs/pull/295) Add Pangea4 spack config
- [ ] #3561 + [TPL-296](https://github.com/GEOS-DEV/thirdPartyLibs/pull/296) Add Pangea4 CI Job (spack based)
- [ ] TODO +  [TPL-302](https://github.com/GEOS-DEV/thirdPartyLibs/pull/302) Update Pangea3 CI build to spack with latest available modules (>=gcc11 and >=cuda11.8)
- [x] [TPL-299](https://github.com/GEOS-DEV/thirdPartyLibs/pull/299) Add Pangea3 spack config
- [x] [TPL-257](https://github.com/GEOS-DEV/thirdPartyLibs/pull/257) Add Pangea3 CI Jobs in the github actions
- [x] [TPL-308](https://github.com/GEOS-DEV/thirdPartyLibs/pull/308) Add Pine spack config 
- [x]  #3611 Create Maple host-configs
- [x] [TPL-313](https://github.com/GEOS-DEV/thirdPartyLibs/pull/313) Add Maple spack config 
- [x]  [TPL-316](https://github.com/GEOS-DEV/thirdPartyLibs/pull/316) Add variant for CUDA stack size 
- [ ] Remove host-configs once all spack builds are OK

### PR DESCRIPTION
**Pull request by Bubusch**
Adds the Pangea 4 host config files :

| host-config   | Description | Compiler | MPI | BLAS-LAPACK |
| -------- | ------- | ------- | ------- | ------- |
| pangea4-base.cmake  | base configuration for Pangea4 (CPU cluster) using Hypre, no openMP  | X | X | X |
| pangea4-gcc12.1-hpcxompi2.17.1-onemkl2023.2.0 | default gcc build | cray wrappers for gnu (cc, CC, ftn)  | HPC-X OpenMPI | oneAPI MKL |
| pangea4-gcc12.1-openmpi4.1.6-onemkl2023.2.0 | alternative gcc build | cray wrappers for gnu (cc, CC, ftn)  | OpenMPI | oneAPI MKL |
| pangea4-gcc12.1-openmpi4.1.6-openblas0.3.23 | alternative gcc build | cray wrappers for gnu (cc, CC, ftn)  | OpenMPI | OpenBLAS |
| pangea4-gcc12.1-hpcxompi2.17.1-openblas0.3.23 | alternative gcc build | cray wrappers for gnu (cc, CC, ftn)  | HPC-X OpenMPI | OpenBLAS |

Linked to TPLs' PR [#262](https://github.com/GEOS-DEV/thirdPartyLibs/pull/262).

### CODE REVIEW

File: host-configs/TOTAL/pangea4-base.cmake [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+set( ENABLE_SCOTCH      ON  CACHE BOOL "" FORCE )
+set( ENABLE_SUITESPARSE ON  CACHE BOOL "" FORCE )
+set( ENABLE_TRILINOS    OFF CACHE BOOL "" FORCE )
+set( ENABLE_VTK         ON  CACHE BOOL "" FORCE )
+
+#######################################
+# DEVELOPMENT TOOLS
+#######################################
+
+set( ENABLE_DOXYGEN           ON CACHE BOOL "" FORCE )
```
Comment (MelReyCG) [inline host-configs/TOTAL/pangea4-base.cmake:31]: Why is there no `ENABLE_DOCS` and no `ENABLE_SPHINX`?

File: host-configs/TOTAL/pangea4-base.cmake [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+#######################################
+# PERFORMANCE TOOLS
+#######################################
+
+set( ENABLE_BENCHMARKS ON CACHE BOOL "" FORCE )
+set( ENABLE_CALIPER    ON CACHE BOOL "" FORCE )
+
+#######################################
+# RAJA/CHAI/OPENMP SETUP
```
Comment (MelReyCG) [inline host-configs/TOTAL/pangea4-base.cmake]: As openmp is only used through RAJA, I would name this part "RAJA/CHAI SETUP"

File: host-configs/TOTAL/pangea4-gcc12.1-openmpi4.1.6-onemkl2023.2.0.cmake [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+if( NOT DEFINED ENV{GCC_PATH} )
+    message( FATAL_ERROR "GCC is not loaded. Please load the PrgEnv-gnu/8.4.0 module." )
+endif()
+
+set( CMAKE_C_COMPILER       "cc"  CACHE PATH "" )
+set( CMAKE_CXX_COMPILER     "CC"  CACHE PATH "" )
+set( CMAKE_Fortran_COMPILER "ftn" CACHE PATH "" )
+
+set( COMMON_FLAGS  "-O3 -march=native -mtune=native" )
+set( RELEASE_FLAGS "-DNDEBUG"                        )
```
Comment (MelReyCG) [inline host-configs/TOTAL/pangea4-gcc12.1-openmpi4.1.6-onemkl2023.2.0.cmake]: Why do you need those explicit flags? I thought that `-O3` was already enabled.

File: host-configs/TOTAL/pangea4-gcc12.1-openmpi4.1.6-onemkl2023.2.0.cmake [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+#######################################
+
+# use :
+# - OpenMPI library
+
+set( ENABLE_MPI ON CACHE BOOL "" )
+
+if( NOT "$ENV{LMOD_MPI_NAME}" STREQUAL "openmpi" )
+    message(FATAL_ERROR "OpenMPI is not loaded. Please load the openmpi/4.1.6 module.")
+endif()
```
Comment (MelReyCG) [inline host-configs/TOTAL/pangea4-gcc12.1-openmpi4.1.6-onemkl2023.2.0.cmake:79]: Don't you need the `MPI_X_COMPILER` & `MPIEXEC` variables? Are they defined on the system?

**Review by jhuang2601 — COMMENTED** [review]
Thanks for this update. Now, I can successfully compile TPLs and GEOS with hpcx-ompi, which is being used to run SEAM cases on P4. However, encountered some troubles on compilation with openmpi.

File: host-configs/TOTAL/pangea4-gcc12.1-openmpi4.1.6-onemkl2023.2.0.cmake (hunk 4 above)
Comment (Bubusch) [reply host-configs/TOTAL/pangea4-gcc12.1-openmpi4.1.6-onemkl2023.2.0.cmake:79]: They are set by find_package(MPI REQUIRED) in cmake/blt/cmake/thirdparty/SetupMPI.cmake

File: host-configs/TOTAL/pangea4-base.cmake (hunk 1 above)
Comment (Bubusch) [reply host-configs/TOTAL/pangea4-base.cmake:31]: added ENABLE_SPHINX

File: host-configs/TOTAL/pangea4-base.cmake (hunk 2 above)
Comment (Bubusch) [reply host-configs/TOTAL/pangea4-base.cmake]: changed to RAJA/CHAI SETUP

**Review by MelReyCG — APPROVED** [review]
seems good to me

**Review by jhuang2601 — APPROVED** [review]
Great work! I've successfully compiled GEOS with the default one and completed multiple SEAM cases on P4.

**Review by MelReyCG — APPROVED** [review]

File: host-configs/TOTAL/pangea4-base.cmake [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+#   - OPENMP OFF
+#   - HYPRE ON
+#
+#######################################
+
+#######################################
+# SCIENTIFIC LIBRARIES
+#######################################
+
+set( ENABLE_FESAPI      ON  CACHE BOOL "" FORCE )
```
Comment (untereiner) [inline host-configs/TOTAL/pangea4-base.cmake]: You can remove ENABLE_FESAPI  #2969

**Comment by Bubusch** [comment]
will run integrated tests just before merge (as no rebase line is required)

**Pull request merged**